R/jointScoreTest.R

```diff
@@ -0,0 +1,74 @@
+# Inputs:
+# * nullmod: null model object without variants
+# * G: genotype matrix with variants to estimate jointly
+
+# Outputs:
+# * list with elements: pve, fixef, covar
+jointScoreTest <- function(nullmod, G) {  # # Check rownames/colnames match.
+
+  # Check that all samples in null model are in the genotype matrix.
+  missing_ids <- setdiff(rownames(nullmod$model.matrix), rownames(G))
+  if (length(missing_ids) > 0) {
+    stop("missing samples in genotype matrix!")
+  }
+
+  # Check for missingness in genotype data.
+  # For now, missingness is not allowed.
+  # Eventually, update to handle missingness the same way as assocTestSingle.
+  if (any(is.na(G))) {
+    stop("genotype matrix cannot contain missing data!")
+  }
+
+  # Reorder samples.
+  idx <- match(rownames(nullmod$model.matrix), rownames(G))
+  G <- G[idx, , drop = FALSE]
+
+  # Genotype adjusted for covariates and random effects.
+  Gtilde <- calcGtilde(nullmod, G)
+
+  # Score statistic.
+  GY <- crossprod(Gtilde, nullmod$Ytilde)
+
+  # GPG.
+  GG <- crossprod(Gtilde)
+
+  # Covariance matrix for estimates.
+  betaCov <- solve(GG)
+  rownames(betaCov) <- colnames(betaCov) <- colnames(G)
+
+  # Fixed effect estimates for the variants.
+  beta <- crossprod(betaCov, GY)
+  se <- sqrt(diag(betaCov))
+
+  # Joint test statistic. Note this is the equivalent of the stat being squared,
+  # by convention.
+  Stat.joint <- as.vector(crossprod(GY, beta))
```

Comment:
I don't think this is a problem, but Stat.joint is just a scalar. I think could be as.numeric() instead